### PR TITLE
GT-242: Add an option cli_args to Application init method, when its valu...

### PR DESCRIPTION
**Pull request by htxiong**
...e is an empty str, no any argument will be passed to CLI application.

### CODE REVIEW

File: krux_boto/__init__.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def __init__(self, logger = None, stats = None, parser = None):
         self.___parser = parser or get_parser(description = self.___name)
 
         ### in case we got some of the information via the CLI
-        self.___args    = self.___parser.parse_args()
+        self.___args    = self.___parser.parse_args(args = cli_args)
```
Comment (anishbhuta) [inline krux_boto/__init__.py]: Please remove the extra spaces before and after the equals sign. This should be done for you automatically if you use 'reformat code' in IntelliJ.

File: krux_boto/__init__.py [hunk 2]
```diff
@@ -157,14 +157,18 @@ def wrapper(*args, **kwargs):
 
 class Application(krux.cli.Application):
 
-    def __init__(self, name = NAME):
+    def __init__(self, name = NAME, cli_args = None):
```
Comment (anishbhuta) [inline krux_boto/__init__.py]: extra spaces before and after equals sign

File: krux_boto/__init__.py [hunk 3]
```diff
@@ -157,14 +157,18 @@ def wrapper(*args, **kwargs):
...
         ### Call to the superclass to bootstrap.
-        super(Application, self).__init__(name = name)
+        super(Application, self).__init__(name = name, cli_args = cli_args)
```
Comment (anishbhuta) [inline krux_boto/__init__.py]: extra space before and after equals sign

File: krux_boto/__init__.py [hunk 4]
```diff
@@ -157,14 +157,18 @@ def wrapper(*args, **kwargs):
...
 
+
+        # if cli_args = None the list of args from sys.argv will be used,
+        # and if its '', no any args will be used.
         self.boto = Boto(
-            parser = self.parser,
-            logger = self.logger,
-            stats  = self.stats,
+            parser   = self.parser,
```
Comment (anishbhuta) [inline krux_boto/__init__.py]: please use code reformat here.

File: krux_boto/__init__.py (hunk 1 above)
Comment (htxiong) [inline krux_boto/__init__.py]: Hi @anishbhuta, I know we should not leave space between variable and = in argument assignment. And I did this for 2 purposes: 1) follow the code styling used in this class, 2) make less  change as I can, then @jib can have a clear view about what did I modify. 

Then, I will fix the code styling issue for both my code and the existing code in the next commit.

**Comment by anishbhuta** [comment]
I imagine you want to use this with the [other pull request](https://github.com/krux/python-krux-stdlib/pull/10). Are you going to update `requirements.pip` to require the new version of `python-krux-stdlib`?

File: krux_boto/__init__.py (hunk 1 above)
Comment (anishbhuta) [inline krux_boto/__init__.py]: Ok, in that case I'll wait for @jib comments.

**Comment by plathrop** [comment]
@htxiong since you closed the other PR, should this one be closed, too?

**Pull request closed without merging**